umm-lib/src/cmr/umm/iso_smap/granule/orbit.clj

```diff
@@ -112,4 +168,39 @@
        (when end-direction
          (let [ed (subs orbit-str
                         end-direction)]
-           (convert-direction (str/trim (subs ed (inc (.indexOf ed ":")))))))))))
+           (convert-direction (str/trim (subs ed (inc (.indexOf ed ":"))))))))
+      "OrbitCalculatedSpatialDomains"
+      (g/->OrbitCalculatedSpatialDomain
+        (when orbital-model-name
+         (let [o-m-name (subs orbit-str
+                          orbital-model-name
+                          (or orbit-number start-orbit-number stop-orbit-number
+                              equator-crossing-longitude equator-crossing-date-time (count orbit-str)))]
```

Comment:
Make `(count orbit-str)` a var in the `let` block and use it throughout this function.

Reply:
Done.